packages/util/src/serializableObject.ts

```diff
@@ -41,6 +47,11 @@ export const toSerializableObject = (obj: unknown, options: ToSerializableObject
       const value = Buffer.from(arr).toString('hex');
       return { [transformationTypeKey]: 'Buffer', value };
     }
+
+    if (hasInnerError(obj) && innerErrorHasData(obj.innerError)) {
+      obj.innerError.data = toSerializableObject(obj.innerError.data);
```

Comment:
I think this results in serializing innerError as plain object, which means that it won't get `[transformationTypeKey]: 'Error'` and won't run through `getErrorPrototype`. Maybe we could wrap `serializeError` in a new function that does this recursively.